expr/eval.go

```diff
@@ -708,21 +708,22 @@ func (c *Call) Eval(rec *zng.Record) (zng.Value, error) {
 	return c.fn.Call(c.args)
 }
 
-func NewCast(expr Evaluator, typ string) (Evaluator, error) {
+func NewCast(expr Evaluator, styp string) (Evaluator, error) {
 	// XXX should handle alias casts... need type context.
 	// compile is going to need a local type context to create literals
 	// of complex types?
-	vc := LookupValueCaster(typ)
-	if vc == nil {
+	typ := zng.LookupPrimitive(styp)
```

Comment:
Should this do a check to that styp is not a container and return a separate error for this case? I imagine we won't be casting arrays, records, set here?

Reply:
If `styp` is a container, we'll end up three lines below with `fmt.Errorf("cast to %s not implemeneted", styp)`, which seems like the outcome we want.